### PR TITLE
Fix failed faceid clears auth settings

### LINKED ISSUE #9871: Respond to FaceID failure callback
# Problem
When FaceID fails, the user is directed to the password unlock screen with no opportunity to try again.

## Implementation
Respond to callback of FaceID failure

- The Face ID check should not happen on a blank background like it is right now, it should happen on top of the pin pad;
- A failed Face ID should allow to enter the passcode instead or re-trigger the Face ID  check
here's where it's a password, not passcode

Screen Shot 2020-01-22 at 17 43 19

## Acceptance Criteria
FaceID behavior is implemented according to design in [Figma](https://www.figma.com/file/dEIljL7UPbXgsZUA0Q4qlE5E/Onboarding?node-id=5038%3A214)

## Notes
Investigate if the behavior can/should be the same for fingerprint

## Future Steps

- Create multiaccount
- Enable FaceID
- Close app (don't sign out, settings will be reset until this issue is resolved: https://github.com/status-im/status-react/issues/9763#issuecomment-573052216
- Open with FaceID, hold phone away from face - make FaceID fail
- Tap Try FaceID again

### PR DESCRIPTION
**Pull request by Ferossgp**
Fixes #9871

### CODE REVIEW

File: src/status_im/multiaccounts/login/core.cljs [hunk 1]
```diff
@@ -213,16 +213,12 @@
               (chaos-mode/check-chaos-mode)
               (multiaccounts/switch-preview-privacy-mode-flag))))
 
-(defn get-new-auth-method [auth-method save-password?]
-  (if save-password?
```
Comment (Ferossgp) [inline src/status_im/multiaccounts/login/core.cljs:215]: I think I should keep it as when :thinking: - I will check possible regression.

**Comment by hesterbruikman** [comment]
by @simonricoo 
> The "faceid" icon is there but on the first click in just closes the keyboard. Second attempt worked just fine

**Review by yenda — APPROVED** [review]

**Comment by Serhy** [comment]
Looks good to me with iOS (FaceID) and Samsung S8 (TouchID) for the:
- biometric icon (faceId/touchID) next to password input is present when it check is failed (two times in a row and 'Enter password' option offered) or cancelled

but from #9871  what's left is :
• biometric check should happen on the same view you're using to unlock your phone
That's not the case at the moment (both iOS and Android):


VS

Screenshot 2020-02-05 at 17 43 49

If I enabled FaceID and re-open app, or login in to account from login page via tap on biometric icon next to password field (when FaceID failed) bio check happens on blank screen. @Ferossgp could we 
make it transparent ( from [Figma](https://www.figma.com/file/dEIljL7UPbXgsZUA0Q4qlE5E/Onboarding?node-id=5038%3A214): ``` background: #000000;
opacity: 0.4;```

**Comment by StatusWrike** [comment]
➤ Sergey Chumak commented:

Looks good to me with iOS (FaceID) and Samsung S8 (TouchID) for the:

  * biometric icon (faceId/touchID) next to password input is present when it check is failed (two times in a row and 'Enter password' option offered) or cancelled

but from #9871  what's left is :
• biometric check should happen on the same view you're using to unlock your phone
That's not the case at the moment (both iOS and Android):
234234234234 ( https://user-images.githubusercontent.com/8749671/73858499-28096580-4841-11ea-9fc9-f3fce606320e.gif )

VS

https://user-images.githubusercontent.com/8749671/73858745-90584700-4841-11ea-816d-53db11fdad79.png

If I enabled FaceID and re-open app, or login in to account from login page via tap on biometric icon next to password field (when FaceID failed) bio check happens on blank screen. @Ferossgp could we 
make it transparent ( from Figma ( https://www.figma.com/file/dEIljL7UPbXgsZUA0Q4qlE5E/Onboarding?node-id=5038%3A214 ): `` background: #000000;
opacity: 0.4;``

**Comment by Ferossgp** [comment]
@Serhy the background is hidden because you have enabled "Hide Status previews when app switching" in Privacy and security. The face-id popup is a native app, which is opened above the Status app and changes the focus.

**Comment by churik** [comment]
Don't have phone with FaceID now, so will wait for @Serhy

**Comment by Serhy** [comment]
@Ferossgp you are correct. 
Well, I knew it was "reproducing" even with disabled "Hide Status previews" option, but I thought that I have to see this transparency on app start from it's closed state. It just looking like that FaceID not transparent but it's because when we start the app - the FaceID pop-up overlaps status logo (like there in gif above). 
Which is not the case when login in via FaceID from login screen (with disabled security setting)
Basically it was my fault, all good!
